src/datascience-ui/ipywidgets/kernel/index.ts

```diff
@@ -0,0 +1,263 @@
+        })
+        .catch((ex) => console.error('Failed to render', ex));
+}
+
+let widgetManagerPromise: Promise<WidgetManager> | undefined;
+async function getWidgetManager(): Promise<WidgetManager> {
+    if (!widgetManagerPromise) {
+        // tslint:disable-next-line: promise-must-complete no-any
+        widgetManagerPromise = new Promise((resolve) => WidgetManager.instance.subscribe(resolve as any));
+        widgetManagerPromise
+            .then((wm) => {
+                if (wm) {
+                    const oldDispose = wm.dispose.bind(wm);
+                    wm.dispose = () => {
+                        widgetManagerPromise = undefined;
+                        return oldDispose();
+                    };
+                }
+            })
+            .catch(noop);
+    }
+    return widgetManagerPromise;
+}
+
+async function createWidgetView(
+    widgetData: nbformat.IMimeBundle & { model_id: string; version_major: number },
+    element: HTMLElement
+) {
+    const wm = await getWidgetManager();
+    try {
+        return await wm?.renderWidget(widgetData, element);
+    } catch (ex) {
+        // tslint:disable-next-line: no-console
+        console.error('Failed to render widget', ex);
+    }
+}
+
+// tslint:disable-next-line: no-any
+function initialize() {
+    try {
+        // Setup the widget manager
+        const postOffice = new PostOffice();
+        const mgr = new WidgetManagerComponent(postOffice);
+        // tslint:disable-next-line: no-any
+        (window as any)._mgr = mgr;
+    } catch (ex) {
+        // tslint:disable-next-line: no-console
+        console.error('Ooops', ex);
+    }
+}
+
+function convertVSCodeOutputToExecutResultOrDisplayData(
+    request: NotebookOutputEventParams
+): nbformat.IExecuteResult | nbformat.IDisplayData {
+    // tslint:disable-next-line: no-any
+    const metadata: Record<string, any> = {};
+    // Send metadata only for the mimeType we are interested in.
+    const customMetadata = request.output.metadata?.custom;
+    if (customMetadata) {
+        if (customMetadata[request.mimeType]) {
+            metadata[request.mimeType] = customMetadata[request.mimeType];
+        }
+        if (customMetadata.needs_background) {
+            metadata.needs_background = customMetadata.needs_background;
+        }
+        if (customMetadata.unconfined) {
+            metadata.unconfined = customMetadata.unconfined;
+        }
+    }
+
+    return {
+        data: {
+            [request.mimeType]: request.output.data[request.mimeType]
+        },
+        metadata,
+        execution_count: null,
+        output_type: request.output.metadata?.custom?.vscode?.outputType || 'execute_result'
+    };
+}
+
+// Create our window exports
+// tslint:disable-next-line: no-any
+(window as any).ipywidgetsKernel = {
+    renderOutput,
+    disposeOutput
+};
+
+// tslint:disable-next-line: no-suspicious-comment
+// This is to workaround bug: https://github.com/microsoft/vscode/issues/110338
```

Comment:
Per linked bug this looks to be fixed. Can we remove? Looks hacky otherwise.

Reply:
Going to leave here just in case but I'll change the comment.